### PR TITLE
Creating Mbed OS and Mbed baremetal targets

### PR DESCRIPTION
**Pull request by 0xc0170**
### Summary of changes 

Mbed OS and Mbed baremetal targets are both interface. I had to fix few things to make that happen. Tested with K64F on blinky. I can see RTOS being properly recognized and added (247 files vs 224 files). 

### Documentation 

----------------------------------------------------------------------------------------------------------------
### Pull request type 

    [] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results 

    [] No Tests required for this change (E.g docs only update)
    [] Covered by existing mbed-os tests (Greentea or Unittest)
    [] Tests / results supplied as part of this PR
    
    
----------------------------------------------------------------------------------------------------------------
### Reviewers 

----------------------------------------------------------------------------------------------------------------

### CODE REVIEW

**Comment by 0xc0170** [comment]
@hugueskamba @rajkan01 please review

**Comment by 0xc0170** [comment]
I am fixing the rest of the targets, quick find and replace combo

**Comment by 0xc0170** [comment]
Targets updated

**Comment by 0xc0170** [comment]
Removed one more not used CMake code. I found another one but not sure what it should do:

```
# Verify that the selected application OS profile is supported by the Mbed target
#if(NOT MBED_CONF_RTOS_PRESENT)
#     if(NOT "bare-metal" IN_LIST MBED_TARGET_SUPPORTED_APPLICATION_PROFILES)
#         message(FATAL_ERROR
#             "Baremetal profile is not supported for this target")
#     endif()
#endif()
```

This branch would be best to compare to feature-cmake, to see what changes we are bringing in.

**Comment by 0xc0170** [comment]
That should be it from my side, waiting for reviews

File: targets/TARGET_STM/TARGET_STM32F4/TARGET_STM32F439xI/TARGET_WIO_3G/CMakeLists.txt [hunk 1]
```diff
@@ -2,14 +2,14 @@
 # SPDX-License-Identifier: Apache-2.0
 
 target_sources(mbed-core
-    PRIVATE
+    INTERFACE
     ONBOARD_QUECTEL_UG96.cpp
     PeripheralPins.c
     PinNames.h
     system_clock.c
```
Comment (hugueskamba) [inline targets/TARGET_STM/TARGET_STM32F4/TARGET_STM32F439xI/TARGET_WIO_3G/CMakeLists.txt]:
```suggestion
        ONBOARD_QUECTEL_UG96.cpp
        PeripheralPins.c
        PinNames.h
        system_clock.c
```

File: rtos/source/ThisThread.cpp [hunk 2]
```diff
@@ -220,7 +220,7 @@ void ThisThread::sleep_for(uint32_t millisec)
 
 void ThisThread::sleep_for(Clock::duration_u32 rel_time)
 {
-#if MBED_CONF_RTOS_PRESENT
+#ifdef MBED_CONF_RTOS_PRESENT
```
Comment (rajkan01) [inline rtos/source/ThisThread.cpp]: Do we need this change

**Review by rajkan01 — APPROVED** [review]
LGTM

**Comment by rajkan01** [comment]
> 
> 
> Removed one more not used CMake code. I found another one but not sure what it should do:
> 
> ```
> # Verify that the selected application OS profile is supported by the Mbed target
> #if(NOT MBED_CONF_RTOS_PRESENT)
> #     if(NOT "bare-metal" IN_LIST MBED_TARGET_SUPPORTED_APPLICATION_PROFILES)
> #         message(FATAL_ERROR
> #             "Baremetal profile is not supported for this target")
> #     endif()
> #endif()
> ```
> 
> This branch would be best to compare to feature-cmake, to see what changes we are bringing in.


@0xc0170 We need this check in CMakeList.txt and similar check in device/CMakeLists.txt. I will add this check later once PR  https://github.com/ARMmbed/mbed-tools/pull/93 for adding _supported_appliation_profiles_ support in mbedtools gets merged.

File: rtos/source/ThisThread.cpp (hunk 2 above)
Comment (0xc0170) [reply rtos/source/ThisThread.cpp]: Not now, I'll revert

Comment (0xc0170) [reply rtos/source/ThisThread.cpp]: done

**Comment by 0xc0170** [comment]
As already reviewed, I'll squash all into one commit (all are related and together).

**Comment by 0xc0170** [comment]
Rebase in progress

File: CMakeLists.txt [hunk 3]
```diff
@@ -13,9 +13,24 @@ endif()
 include(${MBED_ROOT}/tools/cmake/core.cmake)
 include(${MBED_ROOT}/tools/cmake/profile.cmake)
 
-add_library(mbed-core OBJECT)
-# TODO CMAKE: Replace mbed-os by a new one that will include mbed-core and mbed-rtos
-add_library(mbed-os ALIAS mbed-core)
+add_library(mbed-core INTERFACE)
+
+# Mbed OS = mbed core + mbed rtos
```
Comment (hugueskamba) [inline CMakeLists.txt]:
```suggestion
```
That can be seen in code.

File: CMakeLists.txt [hunk 4]
```diff
@@ -13,9 +13,24 @@ endif()
...
+add_library(mbed-os INTERFACE)
+
+target_link_libraries(mbed-os
+        INTERFACE
+            mbed-rtos
+            mbed-core
+)
+
+# Mbed baremetal = mbed core
```
Comment (hugueskamba) [inline CMakeLists.txt]:
```suggestion
```
That can be seen in code.

File: storage/kvstore/CMakeLists.txt [hunk 5]
```diff
@@ -21,7 +21,7 @@ target_sources(mbed-storage-kvstore
 )
 
 target_link_libraries(mbed-storage-kvstore
-    INTERFACE
+    INTERFACE
```
Comment (hugueskamba) [inline storage/kvstore/CMakeLists.txt]:
```suggestion
    INTERFACE
```

File: CMakeLists.txt (hunk 3 above)
Comment (0xc0170) [reply CMakeLists.txt]: I was thinking if we should do more. I'll remove for now both comments

File: storage/kvstore/CMakeLists.txt (hunk 5 above)
Comment (0xc0170) [reply storage/kvstore/CMakeLists.txt]: fixed

File: CMakeLists.txt (hunk 4 above)
Comment (0xc0170) [reply CMakeLists.txt]: fixed

File: CMakeLists.txt (hunk 3 above)
Comment (0xc0170) [reply CMakeLists.txt]: fixed

**Comment by 0xc0170** [comment]
I rebased, cleaned up the first commit (splitting rtos), so no baremetal included in there.

**Comment by 0xc0170** [comment]
I'll move this to Mbed OS repo, where we can complete the review and integration (testing)

**Pull request closed without merging**